absl/copts.bzl

```diff
@@ -103,7 +103,9 @@ MSVC_FLAGS = [
     "/wd4244",  # conversion from 'type1' to 'type2', possible loss of data
     "/wd4267",  # conversion from 'size_t' to 'type', possible loss of data
     "/wd4800",  # forcing value to bool 'true' or 'false' (performance warning)
+    "/DNOMINMAX",  # Don't define min and max macros (windows.h)
     "/DWIN32_LEAN_AND_MEAN",  # Don't bloat namespace with incompatible winsock versions.
+    "/D_CRT_SECURE_NO_WARNINGS",  # Don't warn about usage of insecure C functions
```

Comment:
I'm uncomfortable about this.  I'd rather mark each `getenv` call with a #warning pragma so we can be explicit about calling unsafe functions.

Reply:
I am not sure if this is practical. There are just too many C functions marked as insecure (`fopen`, `strcpy`, `getenv`, `rand`, `memcpy`, `printf` etc.). Using #warning pragma will be too verbose.

See the full list at https://msdn.microsoft.com/en-us/library/wd3wzwts.aspx

Reply:
Okay, that's fine for now.  I wonder if it's practical to have shims which use teh safe versions provided by msvc and C11 when they're available and fallback to the unsafe versions otherwise.

Reply:
`-D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1` can be used to tell MSVC to automatically convert unsafe CRT function calls to safe CRT functions (only works for certain cases).

The so-called "safe" CRT functions are just "unsafe/normal" CRT functions with extra bound-checks, so they tend to have slightly _worse performance_. Abseil code probably already does enough bound-checks, so explicit "safe" CRT functions aren't useful.

`D_CRT_SECURE_NO_WARNINGS` is a very common flag used by many cross-platform projects, I won't worry too much about it. See https://github.com/llvm-mirror/llvm/blob/master/cmake/modules/HandleLLVMOptions.cmake#L336

See https://msdn.microsoft.com/en-us/library/ms175759.aspx